temporal/temporal.go

```diff
@@ -0,0 +1,101 @@
+package temporal
+
+import (
+	"context"
+	"crypto/tls"
+
+	"github.com/superfly/fly-autoscaler"
+	"go.temporal.io/api/workflowservice/v1"
+	"go.temporal.io/sdk/client"
+)
+
+var _ fas.MetricCollector = (*MetricCollector)(nil)
+
+type MetricCollector struct {
+	name   string
+	client client.Client
+
+	// Host & port of the Temporal server. Defaults to localhost:7233.
+	// Must be set before calling Open().
+	Hostport string
+
+	// Namespace to connect to. Defaults to "default".
+	// Must be set before calling Open().
+	Namespace string
+
+	// Certificate & key data. Optional.
+	// Must be set before calling Open().
+	Cert []byte
+	Key  []byte
+
+	// Query string used to filter running workflows.
+	Query string
+}
+
+func NewMetricCollector(name string) *MetricCollector {
+	return &MetricCollector{name: name}
+}
+
+func (c *MetricCollector) Open() (err error) {
+	opt := client.Options{
+		HostPort:  c.Hostport,
+		Namespace: c.Namespace,
+	}
+
+	if len(c.Cert) != 0 || len(c.Key) != 0 {
+		cert, err := tls.X509KeyPair(c.Cert, c.Key)
+		if err != nil {
+			return err
+		}
+		opt.Credentials = client.NewMTLSCredentials(cert)
+	}
+
+	c.client, err = client.Dial(opt)
+
+	return err
+}
+
+func (c *MetricCollector) Close() error {
+	if c.client != nil {
+		c.client.Close()
+	}
+	return nil
+}
+
+func (c *MetricCollector) Name() string {
+	return c.name
+}
+
+func (c *MetricCollector) CollectMetric(ctx context.Context) (float64, error) {
+	// Append additional query filter, if specified.
+	query := `ExecutionStatus="Running"`
+	if c.Query != "" {
+		query += " AND " + c.Query
+	}
+
+	n, err := c.workflowExecutionN(ctx, query)
+	if err != nil {
+		return 0, err
+	}
+	return float64(n), nil
+}
+
+func (c *MetricCollector) workflowExecutionN(ctx context.Context, query string) (n int, err error) {
+	var nextPageToken []byte
+	for {
+		resp, err := c.client.ListWorkflow(ctx, &workflowservice.ListWorkflowExecutionsRequest{
```

Comment:
At Temporal we don't recommend this approach to scaling (list is eventually consistent and this doesn't help activities in any way), but if you must use workflow count to scale, consider using `CountWorkflow` instead.

Reply:
Thanks, @cretz. I'll take a look at that worker performance guide. I saw the discussion on the KEDA scaler PR and the trade-offs make sense. This implementation was based on one of our customer's existing implementations so I was trying to stick to that at first.

I saw the `CountWorkflow()` call but IIRC it said somewhere that it was only available if you had Elastic set up. I'm just testing with the local Temporal dev server so I figured it wasn't available without a more extensive setup.

Reply:
> This implementation was based on one of our customer's existing implementations so I was trying to stick to that at first.

This won't properly scale for activities.

> I saw the CountWorkflow() call but IIRC it said somewhere that it was only available if you had Elastic set up

In all modern versions of the server, this works regardless of whether you have "advanced visibility" (i.e. Elastic) setup. It should work with any local Temporal setup too.

Reply:
> This won't properly scale for activities.

I'm ok with that for this first cut of the scaler. The user for the original use case has relatively simple workflows so this approach works for them. We can tweak this and improve it in a separate PR.

> It should work with any local Temporal setup too. If not, can you describe which version you're using?

I switched up the code to use `CountWorkflow()` and the basic test seems to run well. Thanks!